### PR TITLE
cava: init program with settings configuration

### PR DESCRIPTION
**Pull request by bddvlpr**
### Description

Added cava program definition that allows automatic generation of the INI configuration file (`.config/cava/config`) with a given attribute set. Camel case attributes will be converted into snake case to abide the configuration properties.

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [X] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/programs/cava.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  options.programs.cava = {
+    enable = mkEnableOption "Cava audio visualizer";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.cava;
+      defaultText = literalExpression "pkgs.cava";
+      description = "Cava package to use";
+    };
```
Comment (sumnerevans) [inline modules/programs/cava.nix:21]:
```suggestion
    package = mkPackageOption pkgs "cava" { };
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Comment by sumnerevans** [comment]
Also needs a news entry

**Comment by bddvlpr** [comment]
My bad, I just noticed the typo, will fix as soon as I’m home again.

**Comment by bddvlpr** [comment]
@sumnerevans Would you mind re-reviewing the changes I've made?

**Comment by bddvlpr** [comment]
@sumnerevans Mentioning again since it's now marked stale.

**Comment by rycee** [comment]
Thanks! I've merged this to master now. Note, I removed the camel case to snake case conversion since we typically avoid doing such transformations when doing RFC42 style settings options.

**Pull request merged**